extension/test/background_test.ts

```diff
@@ -65,26 +87,76 @@ describe('background.ts', function () {
         { config: rcxMain.config }
       );
     });
+
+    it('should throw typeError if tab is undefined', async function () {
+      // }).to.throw(TypeError, 'sender.tab is always defined here.');
+    });
   });
-});
 
-async function sendMessageToBackground({
-  tabId = 0,
-  type,
-  responseCallback = () => {
-    // Do nothing by default.
-  },
-}: {
-  tabId?: number;
-  type: string;
-  responseCallback?: (response: unknown) => void;
-}): Promise<void> {
-  // In background.ts, a promise is passed to `addListener` so we can await it here.
-  // eslint-disable-next-line @typescript-eslint/await-thenable
-  await chrome.runtime.onMessage.addListener.yield(
-    { type: type },
-    { tab: { id: tabId } },
-    responseCallback
-  );
-  return;
-}
+  describe('xsearch', function () {
+    it('should return search results', async function () {
+      const searchStub = sinon.stub().returns(['to eat', 'verb']);
+      rcxMain.search = searchStub;
+      const request = {
+        type: 'xsearch',
+        text: 'た',
+        dictOption: '2',
+      };
+      const sender = { tab: { id: 0 } };
+      const response = sinon.spy();
+
+      // eslint-disable-next-line @typescript-eslint/await-thenable
+      await chrome.runtime.onMessage.addListener.callArgWith(
+        0,
+        request,
+        sender,
+        response
+      );
+
+      expect(searchStub.calledWith(request.text, request.dictOption)).to.be
+        .true;
+
+      expect(response.called).to.be.true;
+    });
+
+    it('should not search if text is empty', async function () {
+      const searchStub = sinon.stub().returns(['to eat', 'verb']);
+      rcxMain.search = searchStub;
+      const request = { type: 'xsearch', text: '', dictOption: '2' };
+      const sender = { tab: { id: 0 } };
+      const response = sinon.spy();
+
+      // eslint-disable-next-line @typescript-eslint/await-thenable
+      await chrome.runtime.onMessage.addListener.callArgWith(
+        0,
+        request,
+        sender,
+        response
+      );
+
+      expect(searchStub).to.not.be.called;
+      expect(response.called).to.be.false;
```

Comment:
I think this one isn't so important (and probably why it's not in the test description).

In fact, I wonder if we should return something to response but it seems like it works fine if it never gets a response but if it got some special response then it's probably fine since the goal is to return early.

Reply:
Honestly, I agree, however it is also nice to have 100% branch coverage isn't it?

Reply:
This shouldn't affect test coverage should it? Was it reported in the coverage report?

IIUC, both of these are in the same statement so by testing that searchStub is not called it's the same as confirming that response is not called.

The goal here is to focus the user's attention on the goal of the underlying code as documented in the test.

If we did care about `response`, my first thought is to make a new test like `should not call response callback if text input is empty` or similar.